### PR TITLE
Add type annotations to the project and run mypy on CI

### PR DESCRIPTION
**Pull request by jdufresne**
Type annotations allow for static type checking to build additional
confidence in the correctness of internal and third party APIs. A type
checker, such as mypy can help discover bugs when adding or refactoring
code.

The package distributes a py.typed file for PEP-561 compliance.
https://mypy.readthedocs.io/en/stable/installed_packages.html#creating-pep-561-compatible-packages

### CODE REVIEW

File: pypinfo/core.py [hunk 1]
```diff
@@ -1,13 +1,17 @@
 import calendar
 import json
 import os
+import re
```
Comment (hugovk) [inline pypinfo/core.py:4]: Unused:
```suggestion
```

**Comment by hugovk** [comment]
How about adding mypy to pre-commit? Then the existing CI pre-commit lint will run it, plus we get pinning to avoid upgrade surprises.

**Comment by jdufresne** [comment]
> How about adding mypy to pre-commit? Then the existing CI pre-commit lint will run it, plus we get pinning to avoid upgrade surprises.

I can go ahead and do this.

One downside is pypinfo will need to repeat its dependencies in the pre-commit configuration so as to pull in their typing information.

**Comment by jdufresne** [comment]
Moved to pre-commit in the latest revision.

**Review by hugovk — APPROVED** [review]
> > How about adding mypy to pre-commit? Then the existing CI pre-commit lint will run it, plus we get pinning to avoid upgrade surprises.
> 
> I can go ahead and do this.
> 
> One downside is pypinfo will need to repeat its dependencies in the pre-commit configuration so as to pull in their typing information.

OK, no strong opinion from me either way, so whichever you think is best. I'll approve anyway :)

**Review by ofek — APPROVED** [review]

**Pull request merged**